cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -365,6 +365,10 @@
     @{HOME}/ r,
     @{HOME}/snap/{,*/,*/*/} rw,
 
+    # experimental
+    @{HOME}/.snap/                rw,
```

Comment:
I'm a bit confused why we need these changes given that we don't have snap-confine changes

Reply:
snap-confine tries to create the user data directory ([source](https://github.com/snapcore/snapd/blob/a79e60b1cb335d7588ca1973f928751498c01ca4/cmd/snap-confine/user-support.c#L27)). I think it usually already exists, so we could just check and skip in that case. However, I'm not sure if there are cases in which it really must be created

Reply:
I see, I added "needs security review" to the PR

Reply:
Should we explicitly deny .snap/auth.json and .snap/warnings.json (anything else?) then?

Reply:
AFAICT, this doesn't give snap-confine access to ~/.snap/\*, only ~/.snap/data/\* (and below), right?

Reply:
yes, afaiu that rule is about accessing the directory itself

Reply:
Yes - the rule `@{HOME}/.snap/                rw,` specifies read-and-write access to the directory named `.snap` (the trailing `/` indicates this is a directory, not a filename). So this allows to list the contents of the directory (ie. `read`) and to mkdir / rmdir the directory (`write`)) but does not provide any access to the contents of the files within the directory itself, so there is no need to add denials for anything else explicitly.